src/jit/importer.cpp

```diff
@@ -3324,7 +3324,7 @@ GenTreePtr Compiler::impIntrinsic(GenTreePtr            newobjThis,
     //
     // We disable the inlining of instrinsics for MinOpts.
     //
-    if (!mustExpand && (opts.compDbgCode || opts.MinOpts()))
+    if (!mustExpand && (opts.compDbgCode || opts.MinOpts()) && !gtIsRecursiveCall(method))
```

Comment:
@jkotas I tried to specially treat recursively called intrinsics by this check, and it can avoid infinite recursion but `IsSupported` always returns `false`. Could you take a look?